### PR TITLE
Increment binding limit by 500 on user gesture (up to 2000)

### LINKED ISSUE #15980: Urgent Please help : Maximum number of bindings reached (1000).
hey guys, we are hitting this error message 
image

Steps to reproduce the error:
1. Please visit : [https://m.aliexpress.com/wholesale/dress.html](https://m.aliexpress.com/wholesale/dress.html) 
2. Open up the `Refine`
3. Choose `Ship To` from the list, select any country and should be hitting the `maximum bindings` within 1 or 2 selections.

ezgif com-video-to-gif 1

This `Ship To` Country List is a combination of 2 lists:
1. we use Java foreach to render for the `Popular Country List`.
2. we then use `amp-list` to fetch and render the remaining Country List based on Alphabetical Orders.
Here is our coding (copy from page source alt + cmd + U) ：

````js
<amp-list credentials="include" layout="fixed-height" src="https://m.aliexpress.com/api/countries/sorted" height="400" width="auto" items="data">
    <template type="amp-mustache">
        <p class="ship-to-title country-category">{{group}}</p>
        {{#countryList}}
        <div class="category-box">
            <p role="button" tabindex="-1" 
             on="tap: AMP.setState({
                refineState: {
                    shipToCountry: '{{code}}'
                },
                shipTo: '{{name}}'
            }),ship-to-wrap.hide, category-wrap.show, attribute-syn.hide, attribute-asyn.show, other-refine-wrap.show, form-filter.submit">
                <span>
                    <span [class]="'{{code}}' ? 'country-flag mright c_{{code}}' : ''"></span>
                    <span>{{name}}</span>
                </span>
                <input type="radio" name="shipTo">
                <span class="radio-button" [class]="(refineState.shipToCountry == '{{code}}') ? 'radio-button active' : 'radio-button'"></span>
            </p>
        </div>
        {{/countryList}}
    </template>
</amp-list>
````
It is just a simple `amp-list` which fetches data and render with `amp-mustache`.

FYI, we are a global website aims to deliver goods to our buyer across the world, and this filter `Ship To` really means a lot to us and our buyers, really do appreciate your guys assistance on this ASAP. Thank you guys.

@choumx @aghassemi

### PR DESCRIPTION
**Pull request by choumx**
Fixes #15980. Partial for #15519.

Quick fix to allow up to 2000 bindings *after* user gestures. Would like to get this into canary cut today.

/to @jridgewell

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-impl.js [hunk 1]
```diff
@@ -144,10 +144,17 @@ export class Bind {
...
+     * The initial value is set to 1000 which, based on ~2ms expression parse
+     * time, caps "time to interactive" at ~2s after page load.
+     *
+     * User interactions can add new bindings (e.g. infinite scroll), so this
+     * can increase over time to a final limit of 2000 bindings.
+     *
      * @private {number}
      */
-    this.maxNumberOfBindings_ = 1000; // Based on ~2ms to parse an expression.
+    this.maxNumberOfBindings_ = 1000;
```
Comment (kristoferbaxter) [inline extensions/amp-bind/0.1/bind-impl.js]: Nit: The value 1000 is the DEFAULT_ALLOWED_BINDING_COUNT, and then references to the value are a little easier to follow below when doing a...

`min(MAX_ALLOWED_BINDING_COUNT, max(DEFAULT_ALLOWED_BINDING_COUNT, current + 500))`

File: extensions/amp-bind/0.1/bind-impl.js [hunk 2]
```diff
@@ -261,7 +268,11 @@ export class Bind {
 
     const expression = args[RAW_OBJECT_ARGS_KEY];
     if (expression) {
-      // Signal on first mutation.
+      // Increment bindings limit by 500 on each invocation to a max of 2000.
+      this.maxNumberOfBindings_ = Math.min(2000,
+          Math.max(1000, this.maxNumberOfBindings_ + 500));
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-impl.js]: The `Math.max` should be unnecessary here.

File: extensions/amp-bind/0.1/bind-impl.js (hunk 1 above)
Comment (choumx) [reply extensions/amp-bind/0.1/bind-impl.js]: Good idea, will address in a follow-up.

File: extensions/amp-bind/0.1/bind-impl.js (hunk 2 above)
Comment (choumx) [reply extensions/amp-bind/0.1/bind-impl.js]: Ah, I used to have this as `Math.max(1000, 1000 + this.numberOfInvocations_ * 500)`. I'll clean this up soon.

**Comment by choumx** [comment]
Thanks for the fast reviews!

**Pull request merged**